src/vuln_analysis/server/webhooks/auth.py

```
@@ -0,0 +1,174 @@
+            )
+        except httpx.HTTPError as exc:
+            raise OAuth2Error(f"Token request to {self._config.auth_token_url} failed: {exc}") from exc
+
+        if response.status_code == 429:
+            # Surface the retry hint without sleeping while the refresh lock is held. The
+            # delivery loop applies its own backoff after this exception releases the lock.
+            retry_after = _parse_retry_after(response.headers.get("Retry-After"))
+            raise OAuth2Error(
+                f"Token endpoint rate-limited (429); retry_after={retry_after:.1f}s",
+                retry_after_seconds=retry_after,
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)